firebolt_provider/hooks/firebolt.py

```diff
@@ -86,36 +107,96 @@ def _get_conn_params(self) -> Dict[str, Optional[str]]:
         conn = self.get_connection(conn_id)
         database = conn.schema
 
-        engine_name = conn.extra_dejson.get("engine_name", None)
+        engine_name = self.engine_name or conn.extra_dejson.get("engine_name", None)
+
+        engine_url = None
+        if engine_name and "." in engine_name:
+            engine_name, engine_url = None, engine_name
+
         account_name = conn.extra_dejson.get("account_name", None)
         conn_config = {
             "username": conn.login,
             "password": conn.password or "",
-            "api_endpoint": conn.host or DEFAULT_API_URL,
+            "api_endpoint": DEFAULT_API_URL,
             "database": self.database or database,
-            "engine_name": self.engine_name or engine_name,
+            "engine_name": engine_name,
+            "engine_url": engine_url,
             "account_name": account_name,
         }
         return conn_config
 
     def get_conn(self) -> Connection:
         """Return Firebolt connection object"""
         conn_config = self._get_conn_params()
-        conn = connect(**conn_config)
+        username, password = conn_config["username"], conn_config["password"]
+        if not (username and password):
+            raise FireboltError("Either username or password is missing")
+
+        conn = connect(
+            auth=UsernamePassword(username=username, password=password),
+            api_endpoint=conn_config["api_endpoint"],
+            database=conn_config["database"],
+            engine_name=conn_config["engine_name"],
+            engine_url=conn_config["engine_url"],
+            account_name=conn_config["account_name"],
+        )
+
         return conn
 
     def get_resource_manager(self) -> ResourceManager:
         """Return Resource Manager"""
         conn_config = self._get_conn_params()
-        return ResourceManager(
+        username, password = conn_config["username"], conn_config["password"]
+        if not (username and password):
+            raise FireboltError("Either username or password is missing")
+
+        manager = ResourceManager(
             Settings(
-                user=conn_config["username"],
-                password=conn_config["password"],
+                auth=UsernamePassword(username=username, password=password),
                 server=conn_config["api_endpoint"],
                 default_region="us-east-1",
             )
         )
 
+        return manager
+
+    def engine_action(self, engine_name: Optional[str], action: str) -> None:
```

Comment:
Can a user instantiate a hook themselves? If so, consider making `_engine_action` "private" and creating `start_engine` and `stop_engine`.
I don't feel strongly about this, but I think this way there's less string passing and it's a bit clearer from the UX perspective.

Reply:
I think, it is not relevant for the user, the user should never use hooks, only available operators